src/App.tsx

```diff
@@ -15,36 +15,104 @@
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT = 'default',
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
+const preparedGoods = [...goodsFromServer];
+
+const sortGoods = (order: SortType, isReversed: boolean): string[] => {
+  let sortedGoods: string[];
+
+  switch (order) {
+    case SortType.ALPHABET:
+      sortedGoods = [...preparedGoods].sort((a, b) => a.localeCompare(b));
+      break;
+    case SortType.LENGTH:
+      sortedGoods = [...preparedGoods].sort((a, b) => a.length - b.length);
+      break;
+    default:
+      sortedGoods = [...preparedGoods];
+  }
+
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>(preparedGoods);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const [sortOrder, setSortOrder] = useState<SortType>(SortType.DEFAULT);
+
+  const handleSort = (order: SortType) => {
```

Comment:
The function name 'handleSort' is good as it starts with a verb, but it could be more descriptive. Consider renaming it to 'handleSortOrderChange' to reflect its purpose more clearly. See checklist item 4.